bigtable/scheduled-backups/src/main/java/com/example/cloud/bigtable/scheduledbackups/CreateBackup.java

```diff
@@ -0,0 +1,92 @@
+ * limitations under the License.
+ */
+
+package com.example.cloud.bigtable.scheduledbackups;
+
+import com.example.cloud.bigtable.scheduledbackups.resource.CreateBackupMessage;
+import com.example.cloud.bigtable.scheduledbackups.resource.PubSubMessage;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminClient;
+import com.google.cloud.bigtable.admin.v2.BigtableTableAdminSettings;
+import com.google.cloud.bigtable.admin.v2.models.Backup;
+import com.google.cloud.bigtable.admin.v2.models.CreateBackupRequest;
+import com.google.cloud.functions.BackgroundFunction;
+import com.google.cloud.functions.Context;
+import java.nio.charset.StandardCharsets;
+import java.text.SimpleDateFormat;
+import java.util.Base64;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.threeten.bp.Instant;
+
+public class CreateBackup implements BackgroundFunction<PubSubMessage> {
+  private static final Logger logger = Logger.getLogger(CreateBackup.class.getName());
+  private static ObjectMapper mapper = new ObjectMapper();
+
+  @Override
+  public void accept(PubSubMessage message, Context context) {
+    if (message != null && message.getData() != null) {
+      logger.info("Trigger event:" + message.getData());
+
+      try {
+        String payload = new String(
+            Base64.getDecoder().decode(message.getData().getBytes(StandardCharsets.UTF_8)),
+            StandardCharsets.UTF_8);
+
+        logger.info("Decoded payload:" + payload);
+        CreateBackupMessage cbMessage = mapper.readValue(payload, CreateBackupMessage.class);
+        logger.info("CreateBackup message:" + cbMessage.toString());
+        logger.info("Submitting the create backup request");
+
+        // Create an admin client
+        BigtableTableAdminSettings adminSettings =
+            BigtableTableAdminSettings.newBuilder()
+            .setProjectId(cbMessage.getProjectId())
+            .setInstanceId(cbMessage.getInstanceId()).build();
+
+        try (
+          BigtableTableAdminClient adminClient = BigtableTableAdminClient.create(adminSettings)) {
+          CreateBackupRequest request =
+            CreateBackupRequest.of(cbMessage.getClusterId(), buildBackupId(cbMessage.getTableId()))
+            .setSourceTableId(cbMessage.getTableId())
+            .setExpireTime(buildExpireTime(cbMessage.getExpireHours()));
+
+          Backup backupDetails = adminClient.createBackup(request);
+
+          logger.info("Submitted backup request :" + backupDetails.getId()
+                          + ": that will expire at:" + backupDetails.getExpireTime());
+        } catch (Exception e) {
+          logger.log(Level.SEVERE, "Caught Exception creating backup:"
+            + e.toString(), e);
+        }
+      } catch (Exception e) {
+        logger.log(Level.SEVERE, "Caught Exception running the create backup function:"
+            + e.toString(), e);
+      }
```

Comment:
Can these be more specific?

Reply:
Do you mean the log message or exception type?

Reply:
I believe the exception type, see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md#exception-handling for guidance